src/compiler/tsc.ts

```diff
@@ -415,7 +415,8 @@ namespace ts {
 
             const compileResult = compile(rootFileNames, compilerOptions, compilerHost);
 
-            if (!compilerOptions.watch) {
+            // Firefox has Object.prototype.watch
+            if (!compilerOptions.watch || !compilerOptions.hasOwnProperty("watch")) {

```

Comment:
Good points!

But if we go that length, it might be cleaner to rename the option to say `watchOption` -- then nobody would have to worry about helper functions, comments and the rest of this complexity. And also simpler code is better performance.

Well, we can still have a wee comment next to the option itself, but that's it.